src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -28,6 +32,53 @@ function formatForExpectedPixels(format: RegularTextureFormat): RegularTextureFo
 }
 
 class F extends CopyToTextureUtils {
+  private anotherProvider: DeviceProvider | undefined;
+  /** Must not be replaced once acquired. */
+  private anotherDevice: GPUDevice | undefined;
+
+  getAnotherDevice(): GPUDevice {
```

Comment:
This ends up being very similar to `selectMismatchedDeviceOrSkipTestCase`. Probably we can just use it instead. How about moving it from ValidationTest to GPUTest and using it here?